### ISSUE TITLE
Support performance/APM features

### ISSUE THREAD

**Issue by anthonator** [getsentry/sentry-elixir#426]
Does this project plan to add support for Sentry's performance/APM features? Elixir's/Erlang's telemetry support seems like a perfect fit for this.

**Comment by mitchellhenke** [comment]
It’s something I’ve definitely been keeping an eye on. Now that it’s released, I’ll be trying to get some proofs of concept together.

**Comment by anthonator** [comment]
Is there anything the community can do to help?

**Comment by mitchellhenke** [comment]
Good question, thanks for asking it!

Once I have a better handle on the structure, I’ll create issues around the features which should be a great place for contributions 🙂

**Comment by mitchellhenke** [comment]
It turns out that before this can happen I will need to make some bigger changes to move existing functionality to the new Unified API and Envelope data format, so I'll be tackling that first.

**Comment by anthonator** [comment]
I wonder if a good MVP would be just to have the ability to send metrics to Sentry. We can manually hook up telemetry since Phoenix, Ecto, etc. have good telemetry support.

**Comment by UnderFenex** [comment]
Waiting for this feature :)

**Comment by mitchellhenke** [comment]
It's an exciting feature that's at the top of my list for support!

Ruby's conversion to the Unified API and Envelope format gives a bit of an insight into what needs to happen first: https://github.com/getsentry/sentry-ruby/projects/3

This library's footprint is smaller, so it should be easier in some respects 🙂

**Comment by hauleth** [comment]
Is there API specification for that? I could review the possibility to implement such feature if only there is stable API for that. I could make it a backend for Erlang's OpenTelemetry implementation.

**Comment by mitchellhenke** [comment]
Apologies this is taking longer than anticipated, but I'm hoping that the holidays will give me some time to finish the partial implementation I have.

@hauleth It's sort of spread across a handful of pages: https://develop.sentry.dev/sdk/event-payloads/transaction/, https://develop.sentry.dev/sdk/event-payloads/span/ have some description, but I'd be very interested in making it easier to integrate with OpenTelemetry.

**Comment by speeddragon** [comment]
Is the APM feature available for free accounts? I don't mind to help to implement this feature, but I was trying to check the API to send an envelope request and I'm always getting 429.

**Comment by mitchellhenke** [comment]
@speeddragon I'm not sure, I'd reach out to Sentry to see.

I've been making slow progress, but not as much as I'd like 🙂

**Comment by speeddragon** [comment]
I'm trying to understand the request that needs to be made, I can make a successful request, but nothing appears on the Performance tab. I've testes with a JavaScript project and it works, I need to check what is missing in my HTTP request. If I make some progress I will reply back.

**Comment by tsloughter** [comment]
Sentry is adopting OpenTelemetry in general, right? It doesn't cover all Sentry features at this time but starting with tracing and using the OpenTelemetry Erlang/Elixir lib for this would mean many more people involved in helping move it forward and a shared base.

You, or anyone reading this interested in these Sentry features for Elixir, should check out the Observability Working Group of the Erlang Ecosystem Foundation https://erlef.org/wg/observability and the Erlang/Elixir OpenTelemetry SIG https://github.com/open-telemetry/community#special-interest-groups

**Comment by eprothro** [comment]
Is there any update on support for tracing? 

We're wanting to migrate from NewRelic to Sentry, and in our POC, we see the `Sentry-Trace` header make it from our front end Next.js project to our back-end elixir/absinthe project, but these error transactions aren't related on the Sentry Server side :/.

**Comment by tsloughter** [comment]
@ananthakumaran nice. If you want it'd be great to get the propagator added to https://github.com/open-telemetry/opentelemetry-erlang-contrib/

**Comment by whatyouhide** [comment]
Closing this in favor of a single issue about APM, that is, #538.
